src/ui/components/UploadScreen/TeamSelect.tsx

```diff
@@ -11,13 +24,36 @@ export default function TeamSelect({
   selectedWorkspaceId: string | null;
   handleWorkspaceSelect: (id: string | null) => void;
 }) {
-  const displayedWorkspaces = [{ id: null, name: "My Library" }, ...workspaces].sort();
+  const [expanded, setExpanded] = useState(false);
+  const personalWorkspace = { id: null, name: "My Library" };
+  const displayedWorkspaces = [personalWorkspace, ...workspaces].sort();
+
+  const handleSelect = (workspace: Workspace | typeof personalWorkspace) => {
+    handleWorkspaceSelect(workspace.id);
+    setExpanded(false);
+  };
+
+  const selectedWorkspaceName = displayedWorkspaces.find(w => w.id === selectedWorkspaceId)!.name;
+  const button = <TeamSelectButton selectedWorkspaceName={selectedWorkspaceName} />;
 
   return (
-    <SelectMenu
-      selected={selectedWorkspaceId}
-      setSelected={handleWorkspaceSelect}
-      options={displayedWorkspaces}
-    />
+    <PortalDropdown
+      buttonContent={button}
+      setExpanded={setExpanded}
+      expanded={expanded}
+      buttonStyle=""
+      distance={0}
+      position="bottom-right"
+    >
+      <Dropdown widthClass="w-64">
+        <div className="max-h-48 overflow-auto">
+          {displayedWorkspaces.map(w => (
```

Comment:
what do you think about calling this `workspace`? `w` works fine when it's this simple, but if it ever grows or gets more variables in it it would be confusing. I usually only use one letter names when that letter is *super* well understood and the function is a fat-arrow, no brackets function. For example:

`onClick={(e) => e.preventDefault()}`

or

`people.map(p => p.name)`

I realize that this is a border-line case so ITYJ.